platform_plugin_communications/tests/test_views.py

```diff
@@ -0,0 +1,208 @@
+    @patch("platform_plugin_communications.api.views.is_bulk_email_feature_enabled")
+    @patch("platform_plugin_communications.api.views.create_course_email")
+    @patch("platform_plugin_communications.utils.get_course_email")
+    @patch("platform_plugin_communications.utils.submit_task")
+    def test_send_email_to_individual_learners(
+        self,
+        mock_submit_task,
+        mock_get_course_email,
+        mock_create_course_email,
+        mock_is_bulk_email_feature_enabled,
+        mock_get_course_overview_or_none,
+    ):
+        """
+        Test case for sending email to individual learners.
+        """
+        from platform_plugin_communications.edxapp_wrapper.instructor_tasks import (  # noqa pylint: disable=import-outside-toplevel
+            InstructorTaskTypes,
+        )
+
+        request = Mock()
+        request.method = "POST"
+        targets = ["myself"]
+        individual_learners_emails = ["student@openedx.org"]
+        subject = "Test Subject"
+        message = "Test Message"
+        extra_targets = {"emails": individual_learners_emails}
+        request.POST = {
+            "extra_targets": json.dumps(extra_targets),
+            "send_to": json.dumps(targets),
+            "subject": subject,
+            "message": message,
+        }
+        course_id = "course-v1:edX+DemoX+Demo_Course"
+        mock_is_bulk_email_feature_enabled.return_value = True
+        mock_get_course_overview_or_none.return_value = Mock()
+        course_email_mock = Mock()
+        mock_get_course_email.return_value = course_email_mock
+        course_email_mock.targets.all.return_value = [Target(x) for x in targets]
+        mock_create_course_email.return_value = course_email_mock
+        mock_submit_task.return_value = Mock()
+        course_key = CourseKey.from_string(course_id)
+
+        response = send_email(request, course_id)
+
+        mock_is_bulk_email_feature_enabled.assert_called_once_with(course_key)
+        mock_create_course_email.assert_called_once_with(
+            course_key,
+            request.user,
+            targets,
+            subject,
+            message,
+            template_name="test_template",
+            from_addr="test@openedx.org",
+        )
+        course_email_mock.targets.all.assert_called_once()
+        email_id = course_email_mock.id
+
+        mock_submit_task.assert_called_once_with(
+            request,
+            InstructorTaskTypes.BULK_COURSE_EMAIL,
+            send_bulk_course_email_to_learners,
+            course_key,
+            {
+                "email_id": email_id,
+                "to_option": targets,
+                "extra_targets": extra_targets,
+            },
+            hashlib.md5(str(email_id).encode("utf-8")).hexdigest(),
+        )
+
+        assert response.status_code == 200
+        assert json.loads(response.content) == {
+            "course_id": course_id,
+            "success": True,
+        }
```

Comment:
[curious] why are we mixing asserts with TestCase methods? is there a good reason for it?

Reply:
Nop